lnbits/core/services.py

```diff
@@ -396,11 +391,51 @@ class PaymentKwargs(TypedDict):
             checking_id="service_fee" + temp_id,
             payment_request=payment_request,
             payment_hash=invoice.payment_hash,
-            pending=False,
+            status=PaymentState.SUCCESS,
         )
     return invoice.payment_hash
 
 
+async def _create_external_payment(
+    temp_id: str,
+    amount_msat: MilliSatoshi,
+    conn: Optional[Connection],
+    **payment_kwargs,
+) -> Payment:
+    fee_reserve_total_msat = fee_reserve_total(amount_msat, internal=False)
+
+    old_payment = await get_standalone_payment(temp_id, conn=conn)
+    if old_payment:
+        if old_payment.pending:
+            raise PaymentError("Payment is still pending.", status="pending")
+        elif old_payment.success:
+            raise PaymentError("Payment already paid.", status="success")
+        elif old_payment.failed:
+            status = await old_payment.check_status()
+            if status.paid:
+                raise PaymentError(
+                    "Failed payment was already paid on the fundingsource.",
+                    status="success",
+                )
+
+    logger.debug(f"creating temporary payment with id {temp_id}")
+    # create a temporary payment here so we can check if
+    # the balance is enough in the next step
+    try:
+        new_payment = old_payment or await create_payment(
```

Comment:
- this is the same as returning `old_payment` at the end of the `if old_payment:` branch
- in other words, `create_payment(...)` will never be called if `old_payment` exists